zebra/zebra_rib.c

```diff
@@ -641,15 +641,56 @@ struct route_entry *rib_lookup_ipv4(struct prefix_ipv4 *p, vrf_id_t vrf_id)
 	return NULL;
 }
 
+struct route_entry *rib_lookup(struct prefix *p, vrf_id_t vrf_id)
```

Comment:
Adding another copy of the route-lookup logic is pretty unappealing. Is this absolutely necessary (and correct)?

Reply:
There is only ipv4 lookup available, no ipv6 lookup was found. changing the prototype of "rib_lookup_ipv4" will lead to code impact as other functions use it. Instead of creating ipv6_lookup, the generic rib_lookup was added.

Reply:
Thanks for this comment. In new version of the PATCH, 
`struct route_entry *rib_lookup_ipv4` will be replaced by 
`struct route_entry *rib_lookup` with generic functionality for ipv4 and ipv6 prefixes with replacement of all occurrences in the code without cut-n-paste.